### PR TITLE
[ENT-7715] feat: parse API Response body of SAPSF to log relevant courses only

### PR DESCRIPTION
**Pull request by justEhmadSaeed**
JIRA Ticket: [ENT-7715](https://2u-internal.atlassian.net/browse/ENT-7715)

Description:
Parse API Response body, received from the SAPSF, extract only the relevant courses to be stored in the API Response Record table for logging purposes.

**Merge checklist:**
- [ ] Any new requirements are in the right place (do **not** manually modify the `requirements/*.txt` files)
    - `base.in` if needed in production but edx-platform doesn't install it
    - `test-master.in` if edx-platform pins it, with a matching version
    - `make upgrade && make requirements` have been run to regenerate requirements
- [ ] `make static` has been run to update webpack bundling if any static content was updated
- [ ] `./manage.py makemigrations` has been run
    - Checkout the [Database Migration](https://openedx.atlassian.net/wiki/spaces/AC/pages/23003228/Everything+About+Database+Migrations) Confluence page for helpful tips on creating migrations.
    - *Note*: This **must** be run if you modified any models.
      - It may or may not make a migration depending on exactly what you modified, but it should still be run.
    - This should be run from either a venv with all the lms/edx-enterprise requirements installed or if you checked out edx-enterprise into the src directory used by lms, you can run this command through an lms shell.
        - It would be `./manage.py lms makemigrations` in the shell.
- [ ] [Version](https://github.com/openedx/edx-enterprise/blob/master/enterprise/__init__.py) bumped
- [ ] [Changelog](https://github.com/openedx/edx-enterprise/blob/master/CHANGELOG.rst) record added
- [ ] Translations updated (see docs/internationalization.rst but also this isn't blocking for merge atm)

**Post merge:**
- [ ] Tag pushed and a new [version](https://github.com/openedx/edx-enterprise/releases) released
    - *Note*: Assets will be added automatically. You just need to provide a tag (should match your version number) and title and description.
- [ ] After versioned build finishes in [GitHub Actions](https://github.com/openedx/edx-enterprise/actions), verify version has been pushed to [PyPI](https://pypi.org/project/edx-enterprise/)
    - Each step in the release build has a condition flag that checks if the rest of the steps are done and if so will deploy to PyPi.
    (so basically once your build finishes, after maybe a minute you should see the new version in PyPi automatically (on refresh))
- [ ] PR created in [edx-platform](https://github.com/openedx/edx-platform) to upgrade dependencies (including edx-enterprise)
    - This **must** be done after the version is visible in PyPi as `make upgrade` in edx-platform will look for the latest version in PyPi.
    - Note: the edx-enterprise constraint in edx-platform **must** also be bumped to the latest version in PyPi.

### CODE REVIEW

**Review by sameenfatima78 — COMMENTED** [review]
Looks good but would be nice if we can add unit tests to test this parsing functionality on different scenarios

File: integrated_channels/sap_success_factors/transmitters/content_metadata.py [hunk 1]
```diff
@@ -19,6 +21,20 @@ def __init__(self, enterprise_configuration, client=SAPSuccessFactorsAPIClient):
...
+        """
+        Filter the response from SAPSF to only include the content
+        based on the content_id
+        """
+        try:
+            parsed_response = json.loads(response)
+            parsed_response["ocnCourses"] = [item for item in parsed_response["ocnCourses"]
+                                             if item["courseID"] == content_id]
+            filtered_response = json.dumps(parsed_response)
+            return filtered_response
```
Comment (zamanafzal) [inline integrated_channels/sap_success_factors/transmitters/content_metadata.py:34]: It would be good if we can capture the exception and log here.

Comment (justEhmadSaeed) [reply integrated_channels/sap_success_factors/transmitters/content_metadata.py:34]: @zamanafzal "LOGGER.error("Error filtering response from SAPSF: %s", exc)" is this good enough for logging?

Comment (zamanafzal) [reply integrated_channels/sap_success_factors/transmitters/content_metadata.py:37]: Any reason we are not using `LOGGER.exception`?

Comment (justEhmadSaeed) [reply integrated_channels/sap_success_factors/transmitters/content_metadata.py:37]: My bad, fixed it.

**Comment by zamanafzal** [comment]
Please upgrade the enterprise version before merging.

**Review by zamanafzal — APPROVED** [review]

**Review by sameenfatima78 — APPROVED** [review]

**Pull request merged**